pkg/devicerepository/store/bleve/create.go

```diff
@@ -0,0 +1,186 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bleve
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/blevesearch/bleve"
+	"github.com/blevesearch/bleve/analysis/analyzer/keyword"
+	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store"
+	"go.thethings.network/lorawan-stack/v3/pkg/devicerepository/store/remote"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/fetch"
+	"go.thethings.network/lorawan-stack/v3/pkg/jsonpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+type indexableBrand struct {
+	BrandPB  string // *ttnpb.EndDeviceBrand marshaled into JSON string
+	ModelsPB string // []*ttnpb.EndDeviceModel marshaled into JSON string
+
+	// Stored in separate fields to support ordering search results
+	BrandID   string
+	BrandName string
+}
+
+type indexableModel struct {
+	BrandPB string // *ttnpb.EndDeviceBrand marshaled into JSON string
+	ModelPB string // *ttnpb.EndDeviceModel marshaled into JSON string
+
+	// Stored in separate fields to support ordering search results
+	BrandID   string
+	BrandName string
+	ModelID   string
+	ModelName string
+}
+
+const (
+	brandsIndexPath = "brandsIndex.bleve"
+	modelsIndexPath = "modelsIndex.bleve"
+)
+
+func newIndex(path string, overwrite bool, keywords ...string) (bleve.Index, error) {
+	mapping := bleve.NewIndexMapping()
+	if st, err := os.Stat(path); err == nil && st.IsDir() && overwrite {
+		if err := os.RemoveAll(path); err != nil {
+			return nil, err
+		}
+	}
+	keywordFieldMapping := bleve.NewTextFieldMapping()
+	keywordFieldMapping.Analyzer = keyword.Name
+	for _, keyword := range keywords {
+		mapping.DefaultMapping.AddFieldMappingsAt(keyword, keywordFieldMapping)
+	}
+	return bleve.New(path, mapping)
+}
+
+func getWorkingDirectory(paths []string) (string, error) {
+	for _, path := range paths {
+		if s, err := os.Stat(path); err == nil && s.IsDir() {
+			return path, nil
+		}
+	}
+
+	// Default to working directory
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(wd, "device-repository"), nil
+}
+
+// Initialize fetches the Device Repository package file and generates index files.
+func (c Config) Initialize(ctx context.Context, fetcher fetch.Interface, overwrite bool) error {
+	wd, err := getWorkingDirectory(c.SearchPaths)
+	if err != nil {
+		return err
+	}
+
+	b, err := fetcher.File("master.zip")
```

Comment:
Wasn't it `package.zip`?

Reply:
Changed to simply retrieve the latest master branch from https://github.com/TheThingsNetwork/lorawan-devices